llms/mlx_lm/tuner/trainer.py

```diff
@@ -287,13 +276,10 @@ def step(batch):
 
             start = time.perf_counter()
 
-        # Save adapter weights if needed
+        # Save adapter weights
         if (it + 1) % args.steps_per_save == 0:
-            checkpoint_adapter_file = (
-                f"{adapter_path}/{it + 1}_{Path(args.adapter_file).name}"
-            )
-            save_adapter(model=model, adapter_file=checkpoint_adapter_file)
-            print(f"Iter {it + 1}: Saved adapter weights to {checkpoint_adapter_file}.")
+            save_adapter(model=model, adapter_file=args.adapter_file)
```

Comment:
Is this going to keep overriding the same adapter file, causing us to lose all the intermediate checkpoint weights?

Reply:
I am thinking maybe we can do something like this:
```
├── lora_adapters/
│   ├── adapter_1/
│   │   ├── adapter_config.json  
│   │   └── adapters.safetensors
│   ├── adapter_2/
│   │   ├── adapter_config.json    
│   │   └── adapters.safetensors
│   ├── adapter_N/
│   |   ├── adapter_config.json    
│   |   └── adapters.safetensors
│   └── adapter_final/
│       ├── adapter_config.json    
│       └── adapters.safetensors
```
It may seem a bit redundant with the `adapter_config.json`, but we would save more precise information for that training iteration. For example, the learning rate of the scheduled learning rate, loss, and anything else related to the training iteration.

Reply:
I went with
```
lora_adapters/
  -adapters.safetensors
  -adapter_config.json
  -00100_adapters.safetensors
...
``

That fits the way we load better from resumed files and saves from needing to pass all the args into the traiing file or hacking a way around copying the config.

Reply:
Alright, maybe we could add an optional adapter file name parameter to allow users to specify a specific checkpoint weight during the inference so they don't have to copy the checkpoint weight into a folder and rename it in order to load it. If you are okay with this approach, I can work on a PR for adding that support.

Reply:
I'm ok with that.. I just didn't think it was a common thing people did?

Reply:
Do you do it yourself? If so, what for?

Reply:
Yeah, maybe not common use cases, but something from my personal experience. It's not always the adapter with the lowest loss that performs the best, so I normally test out a few adapters to see which one is the best. However, I'm happy to just wait and see if this is something people want to add.

Reply:
Do you test them on a downstream task? I wonder if we saved the adapter based on validation loss maybe that would help.